betty/gramps/loader.py

```diff
@@ -1175,3 +1242,432 @@ def _load_attributes_for(
             self._load_attribute_privacy(entity, element, tag)
         if isinstance(entity, HasLinks):
             self._load_attribute_links(entity, gramps_entity_reference, element, tag)
+
+
+@internal
+class GrampsApiLoader(_GrampsLoader):
+    """
+    Load Gramps family history data into a project.
+    """
+
+    @override
+    async def load_xml(self, xml: str) -> None:
+        async with NamedTemporaryFile(mode="w", suffix=".gramps") as temporary_f:
+            with gzip.open(
+                temporary_f.name,  # type: ignore[arg-type]
+                "w",
+            ) as gzip_f:
+                gzip_f.write(
+                    xml.encode("utf-8")  # type: ignore[arg-type]
+                )
+            return await self.load_file(
+                Path(
+                    temporary_f.name  # type: ignore[arg-type]
+                )
+            )
+
+    @override
+    async def load_file(self, file_path: Path) -> None:
+        async with TemporaryDirectory() as temporary_directory_path_str:
+            database: DBAPI = make_database("sqlite")
+            database.load(":memory:")
+            database.set_mediapath(temporary_directory_path_str)
+            database_state = DbState()
+            database_state.db = database
+            user = User()
```

Comment:
Can we intercept any messages printed to the console and redirect them to our logger?